[CS2113T-T12-4] PAC

lowjiayee opened the pull request:
@lowjiayee
@harithadiv
@benchan911
@anqi-nus

JensonWee commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,179 @@
+`EventParser`.
+1. Then, the relevant class that corresponds to the command is created, with the information extracted 
+from the previous step passed into it. It modifies `Event` or `EventList`.
+1. These commands are then returned to `Duke.run()` to `execute()`. 
```
The diagram looks neat and very informative. Perhaps more description would aid the user a better understanding of the diagram.

JensonWee commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,179 @@
+1. Once determined, the relevant class that corresponds to the type of command is created.
+1. Then, the class will execute base on its function. It modifies `AttendanceList`.
+1. These commands are then returned to `Duke.run()` to `execute()`. 
+
```
The diagram shows clearly what is been created when the function is executed. However, description of the attendance class is not detailed enough. Users might wonder where is AttendanceList? Perhaps a few sentence regarding the details of the objects will aid the user a better understanding of the class. Furthermore, what does the grey box represents?

JensonWee replied:
Description of the flow could be more detailed. Perhaps you can include the reason why such methods are implemented as such. The Pros and Cons of the methods and alternative methods.

JensonWee commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,179 @@
+###Calendar Implementation
+
+*Figure 2: Class diagram of the Calendar component*
+
```
Class diagram of the Calendar component is missing from DG. Is this going to be added in the future update? Perhaps telling the user that it would be added in the future would let the user know that it is going to be added.

JensonWee commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,179 @@
+* Any classes (e.g. `Seminar`) that inherit from `Event` class will have similar control flow. 
+
+### Attendance
+![attendance](images/Attendance.png "Class diagram of Attendance component")        
```
The diagram does not display AttendanceList. Is this format correct? Perhaps an indication where AttendanceList is from would aid the user better understanding of the diagram.

JensonWee replied:
Perhaps a sequence diagram will help users to understand the flow of the command execution?

ananda-lye replied:
Missing the AttendanceList class in the diagram. Agrees with previous comments about possibly adding a sequence diagram. Also, I feel that some arrows can be omitted.

JensonWee commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,179 @@
+*Figure 2: Class diagram of the Calendar component*
+
+1. When a user enters a calendar-related command, the command is analysed by `CalendarCommandInterpreter`.
+1. Once determined, the relevant information (eg. semester, academic year) are extracted by `CalendarParser`.
```
Perhaps clearer description of what's 'relevant information' would aid the user in better understanding. Is semester and academic year the only thing that is extracted by the `CalenderParser`?

JensonWee commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,179 @@
+![attendance](images/Attendance.png "Class diagram of Attendance component")        
+*Class diagram of the Attendance component*
+1. When a user enters an attendance-related command, the command is analysed by `AttendanceCommandInterpreter`. 
+1. Once determined, the relevant class that corresponds to the type of command is created.
```
Perhaps clearer description of what's 'relevantclass' would aid the user in better understanding.

JensonWee commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,179 @@
+1. Then, either AddFirstSemester or AddSecondSemester class that corresponds the semester number is created. 
+1. Subsequently, it separates events by the required month and year in `CalendarList`
+1. These commands are then returned to `Duke.run()` to `execute()`. 
+
```
Description of the flow could be more detailed. Perhaps you can include the reason why such methods are implemented as such. The Pros and Cons of the methods and alternative methods.

andy-aw-why commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,179 @@
+{To be added in future revisions}
+
+### Command
+![Command](images/Command.png "Class diagram of Command component")         
```
Is it possible to resize or restructure this diagram such that it is bigger?

sliu107 replied:
Is it possible to reorder the commands in the diagram so that the commands corresponding the same feature can be put together which makes it easier for the readers to understand?

sliu107 replied:
And I think omitting the constructors or some less important methods can simplify the commands diagram. In this way, it will be more easier for you to resize or restructure the diagram.

ananda-lye replied:
May be better to redraw the commands separately

thanhduc2000 commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,179 @@
+{To be added in future revisions}
+
+### Command
+![Command](images/Command.png "Class diagram of Command component")         
+*Class diagram of the Command component*  
```
The diagram is hard to follow since the image is compressed and not viewable in the website and also not zoomable. Moreover, you are using an auto generated diagram from the system, which is not recommended

SibingWu replied:
I suppose it would be better if the diagram is not so clustered? Also it seems that the IDE 0generated diagram is not encouraged for project submission

andy-aw-why commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,179 @@
 
+## Feature Design and Implementation 
+### Event
+![event](images/event.png "Class diagram of Event component")           
```
Hi, is the format for this diagram (and other diagrams as well in this guide) correct? I think it is missing the private/public field for variable and method name.

andy-aw-why replied:
Also, is this generated using SimpleUML plugin for intellij? I don't think it is recommended to auto generate the UML diagrams.

![](https://files.gitter.im/nus-cs2113-AY1920S2/Tutorial-T13-WeiXiang/szcN/image.png)

ananda-lye replied:
Might be better to redraw with the notations that we are more familiar with. Currently there are many arrows, some may be combined to reduce clutter.

thanhduc2000 commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,179 @@
+## Feature Design and Implementation 
+### Event
+![event](images/event.png "Class diagram of Event component")           
+*Class diagram of the Event component*
```
This diagram is informative and easier to follow. However, since this is auto-generated, there are some aspects that auto generated diagrams cannot express some terms in words like accessibility in methods (public, private, etc), and they refer to icons instead

Shannonwje commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,179 @@
+* Any classes (e.g. `Seminar`) that inherit from `Event` class will have similar control flow. 
+
+### Attendance
+![attendance](images/Attendance.png "Class diagram of Attendance component")        
```
AttendanceList can't be seen inside the diagram, is this format correct? Maybe including AttendanceList would make understanding the diagram clearer, it can be in another diagram if AttendanceList can't fit into the current diagram or make the current diagram less neat.

Shannonwje replied:
Also, there is a grey box in some of the classes while it is not in some other classes. Is there some information missing or omitted in the diagram? Perhaps an explanation for the grey boxes would make it less confusing for the readers trying to understand the diagram

Shannonwje commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,179 @@
+1. When a user enters an event-related command, the command is analysed by `EventCommandInterpreter`. 
+1. Once determined, the relevant information (e.g. index, name, time, date, venue) are extracted by 
+`EventParser`.
+1. Then, the relevant class that corresponds to the command is created, with the information extracted 
```
Diagram and explanation of the overall design and implementation looks neat but maybe giving an example of what you mean by relevant classes would make the understanding of the diagram better.

ananda-lye replied:
Unsure of what the relevant class refers to. Can be phrased more explicitly maybe.

thanhduc2000 commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,179 @@
+* Any classes (e.g. `Seminar`) that inherit from `Event` class will have similar control flow. 
+
+### Attendance
+![attendance](images/Attendance.png "Class diagram of Attendance component")        
```
Should add AttendanceList alo since it is covered in ListAttendance

Shannonwje commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,179 @@
+1. Then, the class will execute base on its function. It modifies `AttendanceList`.
+1. These commands are then returned to `Duke.run()` to `execute()`. 
+
+###Calendar Implementation
```
The header for Calendar Implementation seems to be off, it appears as "###Calendar Implementation" rather than an actual header of the section.

Shannonwje replied:
One change would be to make it "### Calendar Implementation" with a space in between instead.

Shannonwje replied:
Another suggestion would be to be consistent with the header names, to be "### Calandar" to be consistent with the header names before like "### Event" and "### Attendance"

Shannonwje commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,179 @@
+
+###Calendar Implementation
+
+*Figure 2: Class diagram of the Calendar component*
```
The class diagram for the calendar component seems to be missing, will it be added in the  future? If it is currently not ready then maybe can consider not referencing to it now to avoid confusion for the readers.

thanhduc2000 reviewed (commented):
Overall, some class diagrams can be improved. All diagrams should not be auto-generated and your team could use drawing tool online instead. Moreover, you could start working on sequential diagrams for easier explanation

Shannonwje commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,179 @@
+* `acadamic year` is parsed into corresponding to only one year according to the semester in `EventParser` class.
+* Calendar view of the whole year is not available. Only semester 1 or 2 of an academic year can be viewed at a time.
+* Event name size must be less than 10 characters to be displayed neatly (current implementation), however
+it can be implement to truncate longer names to fit nicely
```
Overall Design and Implementation is neat and relatively easy to understand. Perhaps can consider adding in examples with sample inputs of how each implementation would work rather than simply explaining the flow.

Shannonwje replied:
Another suggestion to help the reader understand more about your choice of design and implementation would be stating the different methods of implementation, weighing the pros and cons and stating the reason for choosing a certain implementation over another.

Shannonwje reviewed (commented):
Overall DG is clear and well-explained. Some minor errors in formatting of the headers and a missing diagram but overall easy to understand.

SibingWu commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,179 @@
+1. Then, the class will execute base on its function. It modifies `AttendanceList`.
+1. These commands are then returned to `Duke.run()` to `execute()`. 
+
+###Calendar Implementation
```
![image](https://user-images.githubusercontent.com/49228945/77993506-e08fe900-735a-11ea-86b1-22877d0e8f8c.png)
It seems that the title here is not working well. Maybe consider to add one more space before "Calender Implementation"? i.e. `### Calendar Implementation` instead of "###Calendar Implementation"

ananda-lye replied:
Seems to have some formatting error

SibingWu commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,179 @@
+* Any classes (e.g. `Seminar`) that inherit from `Event` class will have similar control flow. 
+
+### Attendance
+![attendance](images/Attendance.png "Class diagram of Attendance component")        
```
It seems that `AttendanceList` and `EventList` are missing in this diagram?

SibingWu replied:
Consulted the prof: "It's OK to omit less important classes from the diagram"

SibingWu commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,179 @@
+1. Once determined, the relevant class that corresponds to the type of command is created.
+1. Then, the class will execute base on its function. It modifies `AttendanceList`.
+1. These commands are then returned to `Duke.run()` to `execute()`. 
+
```
Maybe you can add a sequence diagram here to illustrate the execution?

SibingWu commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,179 @@
 
-{Describe the value proposition: what problem does it solve?}
+Note that:
+* studentList-related commands can be executed without the existence of events.
```
I am not sure if you format this on purpose, but it seems that those parts look similar to `Event`, `Attendance`, and `Calendar Implementation`. Maybe these two parts also belong to `Feature Design and Implementation` section?

sliu107 commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,179 @@
+ 
+### Parser
+*Class diagram of the Parser component*  
+There are total of four Parser classes as shown below. Each Parser class correspond to a feature 
```
It seems the diagram of parser compont is missing here.If you are going to add it in the feature, I think you can draw a class diagram here and add a sequence diagram to give an example at the end of this part.

SibingWu commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,179 @@
+
+1. When a user enters a calendar-related command, the command is analysed by `CalendarCommandInterpreter`.
+1. Once determined, the relevant information (eg. semester, academic year) are extracted by `CalendarParser`.
+1. Then, either AddFirstSemester or AddSecondSemester class that corresponds the semester number is created. 
```
Just minor format thing: maybe better to emphasize the two classes? `AddFirstSemester` or `AddSecondSemester`

SibingWu reviewed (commented):
Structures are generally good and clear. It would be better if you can add more sequence diagrams and replace the IDE-generated diagram

sliu107 commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,179 @@
-## Design & Implementation
+### UI
+![Ui](images/Ui.png "Class diagram of Ui component")                
+*Class diagram of the UI component*   
```
Is it possible to show the high level association between UI and command class in the diagram? Or just add some description in words so that the reader can more easily understand how the ui is used in other classes and how ui be implemented.

sliu107 reviewed (commented):
All the diagrams are class diagrams, I think adding some sequence diagrams is helpful for readers to understand how each part work and how they are connected to each other.

ananda-lye reviewed (commented):
Maybe the generated diagrams can be used as reference only. As other reviewers have pointed out, the notation may not be entirely correct.  Also, consider adding some sequence diagrams which will greatly aid in understanding how the classes interact with one another. Cheers ! :)

benchan911 committed:
Fix Appendix Format & Edited Non-functional Requirements.

lowjiayee committed:
Add Storage to DG

lowjiayee committed:
Merge pull request #123 from benchan911/Benjamin-DeveloperGuideUpdate

Reformat the Developer Guide to improve Readability: Added Content Page

lowjiayee committed:
Force change filename from uppercase to lowercase

lowjiayee committed:
Merge branch 'master' into JiaYee-Guide

lowjiayee committed:
Force PACException.java to PacException.java

lowjiayee committed:
Remove 'exam' and 'tutorial' from help command

lowjiayee committed:
Now print error messages

lowjiayee committed:
Change sout to UI.display

lowjiayee committed:
Change ui.displayMessage() to UI.display()

Also deleted ui.displayMessage()

lowjiayee committed:
Fix UI.display() not printing

lowjiayee committed:
Visual fixes (#122)

* Replace System.out.println to display

* Add list and delete to Event help

* Use StringBuilder in printGetHelp()

* Explain 'relevant class' better

* Add blank lines for better visual

* Change "Duke" to "PAC"

* Add Architecture in DG

* Change 'PAC' to 'Pac'

* Delete StudentListStorageParser (implemented in StudentList)

* Add Storage to DG

* Force change filename from uppercase to lowercase

* Force PACException.java to PacException.java

* Remove 'exam' and 'tutorial' from help command

* Now print error messages

* Change sout to UI.display

* Change ui.displayMessage() to UI.display()

Also deleted ui.displayMessage()

* Fix UI.display() not printing

lowjiayee committed:
Merge branch 'master' into JiaYee-Storage

lowjiayee committed:
Fix closing an uninitialised FileWriter

lowjiayee committed:
Modify message printed when checking if a file exists

lowjiayee committed:
Add status message to show that data is loaded

lowjiayee committed:
Show status message that data is saved

lowjiayee committed:
Change expected output in text-ui-test

lowjiayee committed:
Merge branch 'master' into JiaYee-Event

lowjiayee committed:
Add storage files to .gitignore

lowjiayee committed:
Use more user-friendly exception messages in EventParser

lowjiayee committed:
EventCI and SeminarCI now show more user-friendly exception messages

lowjiayee committed:
CommandInterpreter now shows more user-friendly exception message

lowjiayee committed:
Break up a long string

lowjiayee committed:
Refactor to show the flow of EventParser better

lowjiayee committed:
Do not parse if command type is 'list'

benchan911 committed:
Update UserGuide format (#153)

benchan911 committed:
Added Attendance Parser(#155)

To give the user a choice to use line-by-line prompt or not

benchan911 committed:
Student List, Attendance - Student Attendance Link (#157)

* If a new list is created for attendance, it will update the student list collection

* If there is a similar name that exist, it will append a () to the end of the name.

benchan911 committed:
Student List - Student Command to Student.List (#154)

* Change the command name from student to student.list for clarity

* Update studentListCollection meaning

benchan911 committed:
Attendance - status bug (#152)

* Fixed Attendance Add Message

* Update UserGuide.md

* Fix style

benchan911 committed:
Student List - No Duplicate (#159)

* Enhancement - Does not allow duplicated List Name to be found in student list collection

* Prevent duplicated names in student list

Adding trim() when reading the student list name or student name to remove whitespaces

* No duplicate in attendance list

* Fix the not added message

* fix style

benchan911 committed:
Student List - Student Find Speed Up (#156)

* Change StudentListFind function, it will list before asking to state the name of hte studentlist

* Find student is no longer CASE SENSITVIVE

harithadiv committed:
Event editing bugs fixed and datetime not set to localtime anymore

lowjiayee committed:
Merge branch 'master' into JiaYee-Event

lowjiayee committed:
Merge pull request #160 from lowjiayee/JiaYee-Event

Event-related fix

harithadiv committed:
Fix merge conflict and fix case sensitive event name duplicate

harithadiv committed:
Fix calendar bug due to datetime change + fix case sensitive calendar command

lowjiayee committed:
Merge pull request #158 from lowjiayee/JiaYee-Storage

Storage-related fix

Anqi-nus committed:
Add Test.

Anqi-nus committed:
Change addToList() method in AddPerformance Command from public to private.

Anqi-nus committed:
Merge pull request #163 from Anqi-nus/Anqi-PerformanceTest

Add Performance Test

lowjiayee committed:
Merge pull request #162 from harithadiv/branch-Calendar

Event editing bugs fixed and datetime not set to localtime anymore

Anqi-nus committed:
Fix bug, Performance add is case sensative for event name.

harithadiv committed:
Refactor method name in calendarList

Anqi-nus committed:
Add edit. Ensure it's not case sensitive.

Anqi-nus committed:
Merge branch 'master' into Anqi-Edit

Anqi-nus committed:
Merge branch 'master' of https://github.com/Anqi-nus/tp into Anqi-Edit

# Conflicts:
#	src/main/java/seedu/ui/UI.java

Anqi-nus committed:
Merge branch 'Anqi-Edit' of https://github.com/Anqi-nus/tp into Anqi-Edit

Anqi-nus committed:
Add edit to Help.

Anqi-nus committed:
Add back printInvalidDateTimeFormat

Anqi-nus committed:
Use more assertions.

Anqi-nus committed:
Check empty.

Anqi-nus committed:
Revert "Add back printInvalidDateTimeFormat"

This reverts commit c2199c880d9c67f7b71c0de1bd2d45ffcff7e06c.

Anqi-nus committed:
Revert "Use more assertions."

This reverts commit 7a0361040a1b1421431dc195041b67548a726910.

Anqi-nus committed:
Revert "Check empty."

This reverts commit 9250c2dcaecb28e48744be2f42774d7ddfbaf5b0.

Anqi-nus committed:
Change back.

Anqi-nus committed:
Merge pull request #164 from Anqi-nus/Anqi-Edit

Anqi - editPerformance

lowjiayee committed:
Add PPP

lowjiayee committed:
Merge branch 'master' into JiaYee-Guide

lowjiayee committed:
Update class diagram of Event

harithadiv committed:
Move calendar section above, update target audience section, update FAQ and update command summary

harithadiv committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-T12-4/tp into branch-Calendar

harithadiv committed:
Add test for CalendarCommandInterpreter

harithadiv committed:
add static final strings for error messages in calendar

harithadiv committed:
Add static final strings for event related classes and add help function for calendar

lowjiayee committed:
Update architecture

lowjiayee committed:
Update event.png to Event.png

Anqi-nus committed:
Update UI and change list to view.

harithadiv committed:
Add images to startup using cli

benchan911 committed:
Attendance SLAP (#161)

* SLAP for Attendance class

* SLAP hard for AttendanceList

* SLAP hard commands

* Stremline sortType

* inserted commandType.toLowerCase().trim()

* fix style problem

* Fix addAttendanceList ui portion

harithadiv committed:
Cleanup startup procedures in UG

harithadiv committed:
Merge pull request #165 from lowjiayee/JiaYee-PPP

Add PPP

harithadiv committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-T12-4/tp into branch-Calendar

benchan911 committed:
loadFromStorage for attendanceLIst

harithadiv committed:
Merge pull request #166 from harithadiv/branch-Calendar

Edit UG

harithadiv committed:
Add correct pics to startup

harithadiv committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-T12-4/tp into branch-Calendar

lowjiayee committed:
Merge pull request #170 from benchan911/Benjamin-AddAttendanceList

loadFromStorage for attendanceList

lowjiayee committed:
Merge branch 'master' into JiaYee-Storage

lowjiayee committed:
Refactor parseStorable()

lowjiayee committed:
Fix javadoc

harithadiv committed:
Add PPP

harithadiv committed:
Merge pull request #172 from harithadiv/branch-Calendar

Change pics of terminal in UG

harithadiv committed:
Add PPP (#174)

benchan911 committed:
PPP template (#176)

benchan911 committed:
updated benchan911 PPP (#175)

Anqi-nus committed:
Update name.

Anqi-nus committed:
Merge pull request #168 from Anqi-nus/Anqi-Edit

Update UI and change list to view.

Anqi-nus committed:
Add PPP. Edit UG. Fix bug in delete peformance, to be case insentive.

Anqi-nus committed:
Update help.

Anqi-nus committed:
Change all performances to performanceList

Anqi-nus committed:
Update DG diagram and DG.

Anqi-nus committed:
Allow the user to exit help function, instead of auto exit after each execution.

Anqi-nus committed:
Delete 'list' case.

Anqi-nus committed:
Update UG

Anqi-nus committed:
Split long line to 2.

Anqi-nus committed:
Merge pull request #178 from Anqi-nus/Anqi-Help

Anqi - Update help, DG diagram and DG

Anqi-nus committed:
Add spacing in between console message and user input.

Anqi-nus committed:
Merge pull request #180 from Anqi-nus/Anqi-Help

Add spacing in between console message and user input.

lowjiayee committed:
Merge branch 'master' into JiaYee-Storage

benchan911 committed:
Fix AttendanceList Bug (#182)

benchan911 committed:
fix student help (#183)

* fix student help

* student.list refactor to studentlist

benchan911 committed:
student.list refactor to studentlist (#191)

Anqi-nus committed:
Update DG diagram.

Anqi-nus committed:
Edit UG.

benchan911 committed:
Benjamin v2.1 error handling (#190)

* remove unnecessary exception for `studentCommandInterpreter`

* remove logger from AttendanceCommandInterpreter

* remove unnecessary exception in attendanceCommandInterpreter

* remove unnecessary Exception

* refactor AddAttendanceList and fix bugs

benchan911 committed:
Merge branch 'master' into JiaYee-Guide

lowjiayee committed:
Initialise attendanceList, simplify method to store datetime

lowjiayee committed:
Merge branch 'master' into JiaYee-Guide

lowjiayee committed:
Change to curly arrows

lowjiayee committed:
Merge branch 'JiaYee-Guide' of github.com:lowjiayee/tp into JiaYee-Guide

lowjiayee committed:
Add sequence diagram to Event

lowjiayee committed:
Add EventSequence diagram

lowjiayee committed:
Show commandType consistently

lowjiayee committed:
Add caveats to diagram

benchan911 committed:
Benjamin v2.1 attendance related & student list JUnit (#193)

* new feature, finding an attendance under a specific event

* added edit feature

* refactor Attendance

* Insert JUnit test for attendance

* add JUnit Test for AttendanceList

* Refactor the edit function for Attendance

* added StudentListTest

* added JUnit for studentListCollection

* Added Logging to AttendanceCommandInterpreter, Do not allow Duplicate for AddByLine in `AddAttendanceList`

* added basic logger

* fix bug for AddAttendanceList and help

* fix `EditAttendance`

lowjiayee committed:
Merge pull request #167 from lowjiayee/JiaYee-Guide

Update DG diagram

lowjiayee committed:
Merge pull request #173 from lowjiayee/JiaYee-Storage

Refactor storage

lowjiayee committed:
Merge branch 'master' into JiaYee-Guide

benchan911 committed:
Matching UserGuide to Attendance and StudentList (#194)

* Match UserGuide With StudentList commands

* typo fixed

* Attendance related to UserGuide

* Fix Content Page

* added edit & find

* added edit and find to content page

* Fixed when studentListCollection is empty

* fixed typo

* remove unnecessary line

* replacing empty student list message

* fixed typo

lowjiayee committed:
Update UG to be consistent with program

lowjiayee committed:
Fix Seminar showing time when it is unset

Anqi-nus committed:
Add DG diagrams.

harithadiv committed:
Minor static to non static changes to calendarParser

Anqi-nus committed:
Update DG.

Anqi-nus committed:
Merge branch 'master' into Anqi-Help

Anqi-nus committed:
Update diagrams.

Anqi-nus committed:
Merge branch 'Anqi-Help' of https://github.com/Anqi-nus/tp into Anqi-Help

Anqi-nus committed:
Update content page

harithadiv committed:
Fix UI static and non-static methods for calendar

harithadiv committed:
Merge with master

Anqi-nus committed:
Update diagram.

Anqi-nus committed:
Merge pull request #192 from Anqi-nus/Anqi-Help

Update DG diagram.

lowjiayee committed:
Update Command class diagram

lowjiayee committed:
Change all 'program' to 'application'

lowjiayee committed:
Merge branch 'master' into JiaYee-Guide

Anqi-nus committed:
Update Help sequence diagram.

benchan911 committed:
Benjamin v2.1 minor fix (#197)

* Match UserGuide With StudentList commands

* typo fixed

* Attendance related to UserGuide

* Fix Content Page

* added edit & find

* added edit and find to content page

* Fixed when studentListCollection is empty

* fixed typo

* remove unnecessary line

* replacing empty student list message

* fixed typo

* refactor viewAttendance

* refactor editAttendance

* fix minor bugs in UserGuide

* Updated DG Appendix E

* update appendix D

* Fix minor bugs

* Fix checkstylemain

harithadiv committed:
update dg

harithadiv committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-T12-4/tp into branch-Calendar

Anqi-nus committed:
Add StudentList feature(add delete clear view) descriptions and sequence diagram in DG.

Anqi-nus committed:
Delete blank lines

Anqi-nus committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-T12-4/tp into Anqi-Help

Anqi-nus committed:
Anqi - studentList DG and diagram (#198)

* Update Help sequence diagram.

* Add StudentList feature(add delete clear view) descriptions and sequence diagram in DG.

lowjiayee committed:
Update UG (#195)

* Update UG to be consistent with program

* Fix Seminar showing time when it is unset

* Update Command class diagram

* Change all 'program' to 'application'

Anqi-nus committed:
Delete Interpreter section in DG. Update class diagram for Performance. Add short description for each feature at the beginning of each feature.

Anqi-nus committed:
Merge branch 'master' into Anqi-Help

Anqi-nus committed:
Merge pull request #199 from Anqi-nus/Anqi-Help

Anqi - DG

lowjiayee committed:
Merge branch 'master' into JiaYee-Guide

lowjiayee committed:
Fix formatting error, and event & storage diagram

lowjiayee committed:
Merge pull request #200 from lowjiayee/JiaYee-Guide

DG formatting

lowjiayee committed:
Merge branch 'master' into JiaYee-Guide

harithadiv committed:
update dg

harithadiv committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-T12-4/tp into branch-Calendar

harithadiv committed:
Set theme jekyll-theme-cayman

lowjiayee committed:
Update UG links and startup diagram

harithadiv committed:
Add calendar details to DG

benchan911 committed:
Benjamin v2.1 DG (#201)

* remove the additional line

* Added for #### 2.6.2 Attendance Command Interpreter

* Updated #### 2.6.5 Student Command Interpreter

* #### Features under Attendance Template added

* Update Attendance in DG

* Added `AttendanceCommandInterpreterFlowChart` and `StudentCommandInterpreterFlowChart`

* AttendanceList -> StudentList

* added attendance.png

* added png

* updated SortStudentList

* FindStudentList

* update attendanceList

* added edit Attendance

* add attendance DG

* added Attendance to DG file

* replace the images

* added description

* update images

* update png

harithadiv committed:
Edit UG

harithadiv committed:
Merge pull request #196 from harithadiv/branch-Calendar

Fix static methods related to Calendar

harithadiv committed:
Merge, fix order of calendar in DG

lowjiayee committed:
Merge branch 'master' into JiaYee-PPP

lowjiayee committed:
Update PPP

lowjiayee committed:
Update UG links and startup diagram (#202)

harithadiv committed:
Update DG v2.1  (#203)

* update dg

* update dg

* Add calendar details to DG

* Edit UG

benchan911 committed:
updated PPP (#205)

lowjiayee committed:
Merge branch 'master' into JiaYee-Event

lowjiayee committed:
Add JavaDoc

lowjiayee committed:
Merge pull request #206 from lowjiayee/JiaYee-Event

Add JavaDoc

lowjiayee committed:
Merge pull request #204 from lowjiayee/JiaYee-PPP

PPP

harithadiv committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-T12-4/tp into branch-Calendar

Anqi-nus committed:
Update UG for add & sort performance, and UI for add performance.

Anqi-nus committed:
Delete unecessary import.

Anqi-nus committed:
Merge pull request #224 from Anqi-nus/Anqi-Documentation

Update UG for add & sort performance, and UI for add performance.

lowjiayee committed:
Merge branch 'master' into JiaYee-Guide

lowjiayee committed:
Update release version to 2.1

harithadiv committed:
Fix calendar error messages

harithadiv committed:
Edit UG calendar to contain clearer instructions

Anqi-nus committed:
Fix bugs in Performance: edit message, add index, delete get name.

lowjiayee committed:
Explains default name and bad input for `event add`

Anqi-nus committed:
Delete space.

Anqi-nus committed:
Change "performance" to "performance list"

Anqi-nus committed:
Merge pull request #228 from Anqi-nus/Anqi-Documentation

Anqi Fix bug

lowjiayee committed:
Fix inconsistent error message

harithadiv committed:
Fix more calendar error message bugs

harithadiv committed:
edit dg - calendar section

lowjiayee committed:
Merge branch 'master' into JiaYee-Storage

harithadiv committed:
Minor tweaks to DG

lowjiayee committed:
Fixed attendance status all set to absent after loading

Anqi-nus committed:
Update UG.

harithadiv committed:
fix minor bugs in ugdg

harithadiv committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-T12-4/tp into branch-Calendar

Anqi-nus committed:
Update UI diagram.

harithadiv committed:
minor bug

harithadiv committed:
edit PPP

lowjiayee committed:
Update Storage, Attendance, name of "Implementation"

lowjiayee committed:
Remove link to parser diagram, and format calendar

Anqi-nus committed:
Format diagrams in DG.

Anqi-nus committed:
Update diagram format.

lowjiayee committed:
Merge pull request #229 from lowjiayee/JiaYee-Guide

UG Update

lowjiayee committed:
Merge pull request #230 from lowjiayee/JiaYee-Event

[Event] Fix inconsistent error message

harithadiv committed:
Merge with master

lowjiayee committed:
Merge pull request #233 from lowjiayee/JiaYee-Storage

Fix wrong attendance status after loading data into app

Anqi-nus committed:
Merge branch 'master' into Anqi-Documentation

Anqi-nus committed:
Merge pull request #234 from Anqi-nus/Anqi-Documentation

Update UG.

benchan911 committed:
[StudentList] Bugs Fixed (#236)

* `studentlist find` grammer #208
`studentlist find` output mismatch #207

* [studentlist] empty name for studentlist is accepted #222

* If studentlistcollection is empty after deletion

benchan911 committed:
[Attendance] Bugs fixed (#232)

* address `Attendance edit` can set name as nothing #220
address `attendance edit` exception thrown with invalid student index #218

* `attendance add` exception when non-integer index is provided #215

* `attendance find` shows duplicate string: "Search Results" #216

* adjust userguide

* added additional instruction

harithadiv committed:
merge with master

harithadiv committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-T12-4/tp into branch-Calendar

harithadiv committed:
Merge pull request #235 from harithadiv/branch-Calendar

UGDG and calendar error message issues

lowjiayee committed:
Merge branch 'master' into JiaYee-Guide

harithadiv committed:
Add PPP pic

lowjiayee committed:
Formatting update

harithadiv committed:
Merge pull request #237 from harithadiv/branch-Calendar

Add PPP pic

lowjiayee committed:
Remove unused imports

lowjiayee committed:
Merge pull request #238 from lowjiayee/JiaYee-Guide

Fix DG formatting and Remove unused imports

benchan911 committed:
PPP updated Benjamin (#239)

* Shrink headerse

* added About Us

* added design consideration for attendance

Anqi-nus committed:
Update format.

Anqi-nus committed:
Update content page.

Anqi-nus committed:
Merge branch 'master' into Anqi-Documentation

Anqi-nus committed:
Merge pull request #240 from Anqi-nus/Anqi-Documentation

Update DG

harithadiv committed:
Edit DG manual testing

Anqi-nus committed:
Increase font size of feature heading.

Anqi-nus committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-T12-4/tp into Anqi-Documentation

harithadiv committed:
Merge pull request #241 from harithadiv/branch-Calendar

Edit DG manual testing

lowjiayee committed:
Merge branch 'master' into JiaYee-Storage

Anqi-nus committed:
Merge pull request #242 from Anqi-nus/Anqi-Documentation

DG documentation

lowjiayee committed:
Allows app to continue running despite corrupted data

lowjiayee committed:
Merge pull request #243 from lowjiayee/JiaYee-Storage

Allows app to continue running despite corrupted data

benchan911 committed:
adding studentCommandInterpreter SequenceDiagram (#244)

* adding studentCommandInterpreter SequenceDiagram

* edit abit of PPP

lowjiayee committed:
Merge branch 'master' into JiaYee-Guide

lowjiayee committed:
Add manual testing instruction for storage

Anqi-nus committed:
Resize image.

Anqi-nus committed:
Update.

lowjiayee committed:
Add manual testing instruction for deleting event

Anqi-nus committed:
Update diagram.

Anqi-nus committed:
Merge pull request #245 from Anqi-nus/master

Resize image.

lowjiayee committed:
Add manual testing instruction for adding event

lowjiayee committed:
Merge branch 'master' into JiaYee-PPP

lowjiayee committed:
Add my face inside

lowjiayee committed:
Update to add manual testing

lowjiayee committed:
Merge pull request #246 from lowjiayee/JiaYee-Guide

Manual testing instruction

lowjiayee committed:
Merge pull request #247 from lowjiayee/JiaYee-PPP

Jia yee PPP

Anqi-nus committed:
Update docs

Anqi-nus committed:
Merge pull request #248 from Anqi-nus/master

Update docs

Anqi-nus committed:
update font size.

Anqi-nus committed:
Merge pull request #249 from Anqi-nus/master

update font size.

benchan911 committed:
Changes for CS2101 (#250)

* documentation bug for sorting fixed

* fix a sentence

* final update

* adding the "." formatting

* remove "by the user"

benchan911 committed:
UG bug (#251)

* documentation bug for sorting fixed

* fix a sentence

* final update

* adding the "." formatting

* remove "by the user"

* grammar
